src/idl_parser.cpp

```diff
@@ -68,9 +68,17 @@ static bool ValidateUTF8(const std::string &str) {
   return true;
 }
 
+static CheckedError CheckDepthGuard(Parser &parser) {
```

Comment:
Yeah, there are too many changes in this on PR. I would reduce the number of clang-tidy checks, maybe even add them one at a time and apply them project wide. That way when reviewing, we just have one tidy check to keep in context to make sure they all make sense.